lib/src/debian.dart

```diff
@@ -0,0 +1,282 @@
+  await _removeDirectory(debianDir);
+}
+
+/// Delete the Directory with path [directory].
+Future<void> _removeDirectory(String directory) async {
+  var result = await Process.run("rm", ["-r", directory]);
+  if (result.exitCode != 0) {
+    fail('Unable to remove the directory\n${result.stderr}');
+  }
+}
+
+/// Scans the PPA [repo] for new packages and updates the
+/// release files, also signing them.
+Future<void> _releaseNewPackage(String repo) async {
+  await _updatePackagesFile(repo);
+  await _updateReleaseFile(repo);
+  await _updateReleaseGPGFile(repo);
+  await _updateInReleaseFile(repo);
+}
+
+/// Create the directory `repo/packageName` and relevant subfolders for the
+/// debian package.
+///
+/// Returns the path of created folder.
+Future<String> _createPackageDirectory(String repo, String packageName) async {
+  String debianDir = p.join(repo, packageName);
+  await Directory('$debianDir/DEBIAN').create(recursive: true);
+  await Directory('$debianDir/usr/local/bin').create(recursive: true);
+  return debianDir;
+}
+
+/// Copy all executable files listed in the map [executables] from the `build` folder
+void _copyExecutableFiles(String debianDir) {
+  final executablePath = p.join(debianDir, "usr", "local", "bin");
+  executables.value.forEach((name, path) {
+    run("cp", arguments: [
+      p.join("build", "$name.native"),
+      p.join(executablePath, name)
+    ]);
+  });
+}
+
+/// Generate the control file for the Debian package.
+void _generateControlFile(String debianDir) {
+  var controlFilePath = p.join(debianDir, "DEBIAN", "control");
+
+  String _updatedControlData = replaceFirstMappedMandatory(
+      controlData.value,
+      RegExp(r'Version: ([0-9].*)'),
+      (match) => 'Version: ${version.toString()}',
+      "Couldn't find a version field in the given CONTROL file.");
+
+  writeString(controlFilePath, _updatedControlData);
+}
+
+/// Scan for new .deb packages in the [repo] and update the `Packages` file.
+Future<void> _updatePackagesFile(String repo) async {
+  // Scan for new packages
+  String output = run("dpkg-scanpackages",
+      arguments: ["--multiversion", "."], workingDirectory: repo);
+  // Write the stdout to the file
+  writeString(p.join(repo, 'Packages'), output);
+  // Force Compress the Packages file
+  run("gzip", arguments: ["-k", "-f", "Packages"], workingDirectory: repo);
+}
+
+/// Generate the Release index for the PPA.
+Future<void> _updateReleaseFile(String repo) async {
+  String output = run("apt-ftparchive",
+      arguments: ["release", "."], workingDirectory: repo);
+  writeString(p.join(repo, 'Release'), output);
+}
+
+/// Sign the Release file with the GPG key.
+Future<void> _updateReleaseGPGFile(String repo) async {
+  run("gpg",
+      arguments: [
+        for (String arg in gpgArgs) arg,
+        "--default-key",
+        (gpgFingerprint.value),
+        "--passphrase",
+        (gpgPassphrase.value),
+        "-abs",
```

Comment:
Short flags are nice for handwritten command-line invocations, but for something in a script file it's clearer to use the full flag names.

Reply:
Oh, okay